modules/doc/content/framework_development/error_warning_messaging.md

```diff
@@ -0,0 +1,61 @@
+# Error, Warning, and Informational Messaging
+
+MOOSE provides several macros/functions for emitting errors, warnings, and informational
+messages:
+
+- `mooseError(args)`: Emits an error with a message formed by concatenating `args`.
+- `mooseDocumentedError(repo_name, issue_num, args)`: Same as `mooseError(args)`, but
+  for errors tied to defects documented in an issue tracker. See [#documented_errors].
+- `mooseAssert(condition, msg)`: In a debug executable, emits an error with message
+  `msg` if the boolean value/expression `condition` is `false`. If not in a debug
+  executable, does nothing.
+- `mooseWarning(args)`: Emits a warning with a message formed by concatenating `args`.
+  If the command-line flag `--error` is supplied, then these warnings are promoted
+  to errors.
+- `mooseDeprecated(args)`: Emits a warning related to a deprecated object/feature.
+  If the command-line flag `--error-deprecated` is supplied, then these warnings
+  are promoted to errors.
+- `mooseInfo(args)`: Emits an informational message formed by concatenating `args`.
+- `mooseDoOnce(statement)`: Executes the statement `statement` only once. Useful
+  for warnings and informational messages that would otherwise occur frequently,
+  such as during residual evaluations or for multiple time steps during a transient.
```

Comment:
```suggestion
```

doonce is already nested in mooseInfo

maybe you want to add mooseInfoRepeated to the list?

Reply:
Yeah we can add that. I just think the name is a misnomer - `mooseInfoOnce` might be better.